pkg/controllers/state/statenode.go

```diff
@@ -70,6 +71,13 @@ func (n StateNodes) Pods(ctx context.Context, c client.Client) ([]*v1.Pod, error
 	return pods, nil
 }
 
+// Disruptable filters StateNodes that are meet the IsDisruptable condition
+func (n StateNodes) Disruptable(ctx context.Context, kubeClient client.Client, recorder events.Recorder) StateNodes {
```

Comment:
I'm noticing a lot of these helper methods pass dependencies around (e.g. kubeclient, recorder). Instead, you could encapsulate them inside of our StateNodes struct to minimize parameter wiring

Reply:
You might even want to place it inside StateNode itself

Reply:
My take: Seems odd that a date object would have something that's more global and unrelated to it like the `client.Client`. Is there precedent for doing things like this in upstream K8s or elsewhere in go? I've generally seen this as an anti-pattern.

Reply:
I'd almost advise that we try and avoid passing through these kinds of global constructs all together and pre-fetch the _data_ and then pass the _data_ through. I think this is what @ellistarn  is suggesting too to some degree. I agree that we want to be careful about how we are passing these things around